src/qt/sendcoinsdialog.cpp

```diff
@@ -889,17 +885,29 @@ void SendCoinsDialog::coinControlUpdateLabels()
     }
 }
 
-SendConfirmationDialog::SendConfirmationDialog(const QString& title, const QString& text, const QString& informative_text, const QString& detailed_text, int _secDelay, QWidget* parent)
-    : QMessageBox(parent), secDelay(_secDelay)
+SendConfirmationDialog::SendConfirmationDialog(const QMessageBox::Icon icon, const QString& title, const QString& text, const QMessageBox::StandardButton yes_button, const QString &yes_button_text, const QMessageBox::StandardButton cancel_button, const QString& informative_text, const QString& detailed_text, int _secDelay, QWidget *parent)
+    : QMessageBox(parent),
+    m_yes_button_text(yes_button_text),
+    secDelay(_secDelay)
 {
-    setIcon(QMessageBox::Question);
+    setIcon(icon);
     setWindowTitle(title); // On macOS, the window title is ignored (as required by the macOS Guidelines).
     setText(text);
     setInformativeText(informative_text);
     setDetailedText(detailed_text);
-    setStandardButtons(QMessageBox::Yes | QMessageBox::Cancel);
-    setDefaultButton(QMessageBox::Cancel);
-    yesButton = button(QMessageBox::Yes);
+    setStandardButtons(yes_button | cancel_button);
+
+    // We need to ensure the buttons have Yes/No roles, or they'll get ordered weird
+    QAbstractButton * const cancel_button_obj = button(cancel_button);
+    removeButton(cancel_button_obj);
+    addButton(cancel_button_obj, QMessageBox::NoRole);
```

Comment:
Why `QMessageBox::NoRole` instead of `QMessageBox::Cancel`?

Reply:
```
    // We need to ensure the buttons have Yes/No roles, or they'll get ordered weird
```

Reply:
That's not very clear; `NoRole` is neither Yes or No.

Reply:
Operating systems have different rules about which button needs to be on the left (I don't care too deeply though). Is that the "weird" ordering you're seeing?

Reply:
https://doc.qt.io/qt-5/qmessagebox.html

NoRole | The button is a "No"-like button.
-- | --

Yes, it has to do with which button is left/right.

Reply:
Oh wait, I miss-read `NoRole` as having no role :-)